ci/tests/general/file_reader.py

```diff
@@ -0,0 +1,448 @@
+        :type debug: Boolean
+        :param hosts: Ip of the nodes
+        :type hosts: List of str
+        :param mode: Search mode
+        :type mode: str
+        :param python_error: Whether only python errors should be checked
+        :type python_error: Boolean
+        :param username: Username of the user to login
+        :type username: str
+        :param password: Password of the user to login
+        :type password: str
+        :param suppress_return: only write to file and not return contents
+        :type suppress_return: Boolean
+        :return: Output of a file as string
+        """
+        # Imports
+        from ci.tests.general.general import General
+        from ci.tests.general.general_storagerouter import GeneralStorageRouter
+        from ci.tests.general.general_network import GeneralNetwork
+        from ovs.extensions.generic.sshclient import SSHClient
+
+        if not start:
+            start = (datetime.today() - timedelta(hours=1)).strftime("%Y-%m-%d %H:%M:%S")
+        if not end:
+            end = datetime.today().strftime("%Y-%m-%d %H:%M:%S")
+        # Clear cache
+        open(LogFileTimeParser.FILE_PATH_REMOTE, 'w').close()
+        if not hosts:
+            hosts = GeneralStorageRouter.get_all_ips()
+        for host in hosts:
+            # Print in front of the commands to capture it using stdout through ssh
+            if mode == 'normal_search':
+                if GeneralNetwork.check_if_local(host):
+                    cmd = "LogFileTimeParser().get_lines_between_timestamps(start='{0}', end='{1}'," \
+                          " paths_to_file={2},search_patterns={3}, debug={4}, host='{5}')" \
+                        .format(start, end, paths_to_file, None, debug, host)
+                else:
+                    cmd = "print LogFileTimeParser().get_lines_between_timestamps(start='{0}', end='{1}'," \
+                          " paths_to_file={2},search_patterns={3}, debug={4}, host='{5}')"\
+                        .format(start, end, paths_to_file, None, debug, host)
+            elif mode == 'error_search':
+                if GeneralNetwork.check_if_local(host):
+                    cmd = "LogFileTimeParser().search_for_errors(start='{0}', end='{1}',paths_to_file={2}, debug={3}," \
+                          " host='{4}', python_error={5})" \
+                        .format(start, end, paths_to_file, debug, host, python_error)
+                else:
+                    cmd = "print LogFileTimeParser().search_for_errors(start='{0}', end='{1}',paths_to_file={2}," \
+                          " debug={3}, host='{4}', python_error={5})"\
+                        .format(start, end, paths_to_file, debug, host, python_error)
+            else:
+                raise ValueError("Mode '{0}' is not supported.'".format(mode))
+            # Create a temporary file
+            copyfile("{0}".format(inspect.stack()[0][1]), LogFileTimeParser.TEMP_FILE_PATH)
```

Comment:
I would wrap this in a method to check if there's enough space as the file ends up in /tmp